.github/workflows/linting.yml

```diff
@@ -0,0 +1,80 @@
+---
+name: Linting
+
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - master
+    tags:
+      - 'v**'
+
+jobs:
+  code-gen:
+    name: Submariner K8s API Code Generation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@master
+      - name: Run validation
+        run: make codegen
+
+  dco:
+    runs-on: ubuntu-latest
+    name: DCO Check
+    steps:
+      - name: Get PR Commits
+        id: 'get-pr-commits'
+        uses: tim-actions/get-pr-commits@master
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: DCO Check
+        uses: tim-actions/dco@master
+        with:
+          commits: ${{ steps.get-pr-commits.outputs.commits }}
+
+  gitlint:
+    name: Gitlint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0
+      - run: make gitlint
+
+  markdown-link-check:
```

Comment:
I wonder if this can be a `make` target instead of a GHA only validation

Reply:
The main issue is that we want this check in the submariner-website repo, and that repo doesn't have the Shipyard shared infra.

Reply:
I'd make it a make target if we can at some point (but I'd say that may be a separate PR while we keep the same name for the linting task), even if we use the GHA-only for the website, or... we can add dapper to the website if we want though..

I'm sometimes miss the opportunity to run that validation locally before sending into the outside woruld

Reply:
Yeah, adding the Shipyard infra to all the repos would be nice in the longer-term. As you said, it would be nice to always have the option to run everything locally.